### PR TITLE
Raise an appropriate exception for unsupported VCS installations

### PR DESCRIPTION
**Pull request by fridex**
## This introduces a breaking change

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by fridex** [comment]
```
  py36-pip190: commands succeeded
  py36-pip200: commands succeeded
  py36-pip201: commands succeeded
  py37-pip190: commands succeeded
  py37-pip200: commands succeeded
  py37-pip201: commands succeeded
  py38-pip190: commands succeeded
  py38-pip200: commands succeeded
  py38-pip201: commands succeeded
  py39-pip190: commands succeeded
  py39-pip200: commands succeeded
  py39-pip201: commands succeeded
  congratulations :)
```

File: micropipenv.py [hunk 1]
```diff
@@ -351,6 +355,16 @@ def _get_requirement_info(requirement):  # type: (ParsedRequirement) -> Dict[str
     # Editable requirements.
     editable = getattr(requirement, "editable", False) or getattr(requirement, "is_editable", False)
 
+    # Check for unsupported VCS.
```
Comment (frenzymadness) [inline micropipenv.py:358]: I am thinking about how to make this check better without duplicates. What about something like:
```python
link = getattr(requirement, "requirement", None) or getattr(requirement, "link", None)
if link and link.startswith(("hg+", "svn+", "bzr+")):
    raise NotSupportedError(…)
```

Comment (fridex) [reply micropipenv.py:358]: Sounds reasonable, I've adjusted the PR in 33ecbf6 :+1:

**Comment by frenzymadness** [comment]
The CI is not happy because the error messages don't match:
```
AssertionError: Pattern
"Non-Git VCS requirement 'svn\\+svn://svn.repo/some_pkg/trunk/#egg=SomePackage' is not supported yet"
does not match
'Non-Git VCS requirement <Link svn+svn://svn.repo/some_pkg/trunk/#egg=SomePackage> is not supported yet'
```

**Pull request merged**